django/core/management/commands/makemigrations.py

```diff
@@ -299,3 +315,122 @@ def all_items_equal(seq):
+        else:
+            questioner = MigrationQuestioner(defaults={'ask_rebase': True})
+
+        for app_label, migration_names in conflicts.items():
+            # Grab out the migrations in question, and work out their
+            # common ancestor.
+            rebase_migrations = []
+            for migration_name in migration_names:
+                migration = loader.get_migration(app_label, migration_name)
+                migration.ancestry = [
+                    mig for mig in loader.graph.forwards_plan((app_label, migration_name))
+                    if mig[0] == migration.app_label
+                ]
+                rebase_migrations.append(migration)
+
+            def all_items_equal(seq):
+                return all(item == seq[0] for item in seq[1:])
+
+            rebase_migrations_generations = zip(*(m.ancestry for m in rebase_migrations))
+            common_ancestor_count = sum(1 for common_ancestor_generation
+                                        in takewhile(all_items_equal, rebase_migrations_generations))
+            if not common_ancestor_count:
+                raise ValueError("Could not find common ancestor of %s" % migration_names)
+            # Now work out the operations along each divergent branch
+            for migration in rebase_migrations:
+                migration.branch = migration.ancestry[common_ancestor_count:]
+                migrations_ops = (loader.get_migration(node_app, node_name).operations
+                                  for node_app, node_name in migration.branch)
+                migration.rebased_operations = sum(migrations_ops, [])
+            # In future, this could use some of the Optimizer code
+            # (can_optimize_through) to automatically see if they're
+            # able to rebase. For now, we always just prompt the user.
+            if self.verbosity > 0:
+                self.stdout.write(self.style.MIGRATE_HEADING("Rebasing %s" % app_label))
+                for migration in rebase_migrations:
+                    self.stdout.write(self.style.MIGRATE_LABEL("  Branch %s" % migration.name))
+                    for operation in migration.rebased_operations:
+                        self.stdout.write("    - %s\n" % operation.describe())
+            if questioner.ask_rebase(app_label):
+                # If they still want to merge it, then write out an empty
+                # file depending on the migrations needing merging.
+                numbers = [
+                    MigrationAutodetector.parse_number(migration.name)
+                    for migration in rebase_migrations
+                ]
+                try:
+                    biggest_number = max(x for x in numbers if x is not None)
+                except ValueError:
+                    biggest_number = 1
+
+                # Get a path to the directory of migration files.
+                module_name, explicit = MigrationLoader.migrations_module(app_label)
```

Comment:
Probably these code to know directory of migration files are not cool. But I don't know a better way.